### DISCUSSION TITLE
837 I Looping issue when 2000C exists

### DISCUSSION THREAD

**Discussion by dineshraghuramaiah** [xlate/staedi#308] [Q&A] [answered]
Hello @MikeEdgar 

Hope you are doing good.

I am facing an issue while looping through a 837 I file. 

Having Subscriber loop without DMG and Patient loop with DMG.

I have defined DMG  optional at subscriber level and required at Patient Level.

 

IMPLEMENTATION_UNUSED_SEGMENT_PRESENT : SEGMENT_ERROR, Error= REF at in segment REF

**Comment by dineshraghuramaiah** [comment]
[new_edi_to_xml_837_I.txt](https://github.com/xlate/staedi/files/9834252/new_edi_to_xml_837_I.txt)
here is the schema

**Comment by dineshraghuramaiah** [comment]
![image](https://user-images.githubusercontent.com/47460384/197070100-ff8ea641-d95e-4f3c-acf6-f8863321f80a.png)

**Comment by MikeEdgar** [comment, marked as answer]
@dineshraghuramaiah - I'm curious why you're thinking the issue is `DMG` when the unused segment in the error is `REF`?

**Reply by dineshraghuramaiah** [reply]
> @dineshraghuramaiah - I'm curious why you're thinking the issue is `DMG` when the unused segment in the error is `REF`?

Since, In my implementation i have DMG minoccurs=0 on 2000B and DMG minoccurs=1 in 2000C loop. When i insert a DMG before REF*SY (in loop 2000B) its going through.

**Reply by MikeEdgar** [reply]
Would you be able to provide a full transaction sample that has the issue? That will help to debug it.

**Reply by dineshraghuramaiah** [reply]
> Would you be able to provide a full transaction sample that has the issue? That will help to debug it.

[TEST_CLAIM_ERROR.txt](https://github.com/xlate/staedi/files/9852609/TEST_CLAIM_ERROR.txt)

**Reply by dineshraghuramaiah** [reply]
any updates @MikeEdgar

**Reply by MikeEdgar** [reply]
@dineshraghuramaiah , I haven't had a chance to review yet. Hopefully today!

**Reply by MikeEdgar** [reply]
@dineshraghuramaiah , the definition of Loop 2010 in the `<transaction>` schema has two `REF`s and the `PER` is in the wrong position as well.

This snippet -
```xml
<loop maxOccurs="3" code="Loop-2010">
	<sequence>
		<segment minOccurs="0" maxOccurs="1" type="NM1"/>
		<segment minOccurs="0" maxOccurs="1" type="N3"/>
		<segment minOccurs="0" maxOccurs="1" type="N4"/>
		<segment minOccurs="0" maxOccurs="2" type="REF"/>
		<segment minOccurs="0" maxOccurs="2" type="PER"/>
		<segment minOccurs="0" maxOccurs="1" type="DMG"/>
		<segment minOccurs="0" maxOccurs="3" type="REF"/>
	</sequence>
</loop>
```
Should instead be -
```xml
<loop maxOccurs="3" code="Loop-2010">
    <sequence>
        <segment minOccurs="0" maxOccurs="1" type="NM1"/>
        <segment minOccurs="0" maxOccurs="1" type="N3"/>
        <segment minOccurs="0" maxOccurs="1" type="N4"/>
        <segment minOccurs="0" maxOccurs="1" type="DMG"/>
        <segment minOccurs="0" maxOccurs="2" type="REF"/>
        <segment minOccurs="0" maxOccurs="2" type="PER"/>
    </sequence>
</loop>
```

**Reply by MikeEdgar** [reply]
Would you mind to please open an enhancement [issue](https://github.com/xlate/staedi/issues) to validate :
1. That a schema `<sequence>` does not contain duplicate segments
2. That `<sequence>`'s in an implementation schema have segments in the same order as the corresponding standard/`<transaction>` schema

**Reply by dineshraghuramaiah** [reply]
Thanks @MikeEdgar , let me rearrange REF and validate.

**Reply by MikeEdgar** [reply]
Sounds good. If it looks fine, please mark this thread as answered as well. Thank you!

**Reply by dineshraghuramaiah** [reply]
Sure @MikeEdgar , and i have sent an email to contact@xlate.io on 9/29 requesting demo for licensed version. I didnt get any reply.
